src/Data/Unrestricted/Internal/Replicator.hs

```diff
@@ -0,0 +1,19 @@
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# LANGUAGE GADTs #-}
+{-# OPTIONS_HADDOCK hide #-}
+
+module Data.Unrestricted.Internal.Replicator (RepStream (..), Replicator (..)) where
+
+import Prelude.Linear.Internal
+import Data.V.Linear.Internal.V (FunN)
+import Prelude (Either(..))
+import GHC.TypeLits
+
+data RepStream a where
+  RepStream :: (s %1 -> a) -> (s %1 -> (s, s)) -> (s %1 -> ()) -> s %1 -> RepStream a
+
+data Replicator a where
+  Moved :: a -> Replicator a
+  CollectFrom :: RepStream a %1 -> Replicator a
```

Comment:
You don't have access to the `Applicative` class at this point†, but you should have a `pure` and a `(<*>)` function nonetheless. For both types.

We will probably need other functions, like a function `Replicator %1 -> RepStream`.

† Sometimes I feel as though this library should all be one big `Internal` file where everything is mutually recursive, and then we use the file hierarchy purely for exports.

Reply:
Random thought: should we use unboxed pairs (and unit) in the projections? Maybe it'll be better?

Reply:
If we use unboxed tuple for the `dups` projector, then it becomes harder to write `dupR` in terms of `dup2` (or we should also change the `dup2` signature to use unboxed tuples).
I don't know how to write an unboxed unit, but if we return some in the `consumes` projector, then writing `consume` (from `Consumable`) for `ReplicationStream` is also less trivial.
I also tried to use unboxed tuples in the `Applicative` instance for `ReplicationStream` (for `<*>`), but AFAIK, it's not possible because the projections cannot receive an unboxed tuple as a function parameter.

Also, without 9.2, I don't see how we can actually write this:
```
dupsf sf' & \case
  (# sf1, sf2 #)
```
We need the `func_call & lambda_case` syntax here for linearity, but because the result of the function call is now passed as an argument to `&`, then unboxed tuples are not allowed in that position.

Reply:
Unboxed unit is `(##)`.

Reply:
Well, let's wait 9.2 before we decide to revisit this issue, then. It's not worth agonising over.